epic_capybara/cli/bara.py

```diff
@@ -143,27 +132,86 @@ def bara(files, match, unmatch, serve):
                         pvalue = 0
                     print(key)
                     print(prev_file_arr, file_arr, f"p = {pvalue:.3f}")
-                    collection_with_diffs.add(branch_name)
+                    collection_with_diffs[branch_name] = min(pvalue, collection_with_diffs.get(branch_name, 1.))
+
+            # Figure
+            h = (
+                Hist.new
+                .Reg(nbins, 0, x_range, name="x", label=key)
+                .Int64()
+            )
+            h.fill(x=ak.flatten(file_arr - x_min, axis=None))
+
+            ys, edges = h.to_numpy()
+            y0 = np.concatenate([ys, [ys[-1]]])
+            legend_label=label + (f"\n{100*pvalue:.0f}%CL KS" if pvalue is not None else "")
+            fig.step(
+                x=edges + x_min,
+                y=y0,
+                mode="after",
+                legend_label=legend_label,
+                line_color=color,
+                line_width=line_width,
+                line_dash=line_dash,
+            )
+            fig.varea_step(
+                x=edges + x_min,
```

Comment:
Probably can avoid information duplication with bokeh's `ColumnDataSource`.

Reply:
But I am worried about the storage size. If we could write a cbor or json.gz and uncompress with js in the browser, that might be more performant too.

Reply:
Yeah, I was only thinking about the x values.

The CompressionStream was something I thought about (we could compress the whole screens). I don't remember why that wasn't pursued. The browser support for it seems decent.